uberfire-extensions/uberfire-commons-editor/uberfire-commons-editor-client/src/main/java/org/uberfire/ext/editor/commons/client/menu/DownloadMenuItemBuilder.java

```diff
@@ -74,6 +75,7 @@ MenuItem makeMenuItem(final String caption,
     }
 
     private String getFileDownloadURL(final Supplier<Path> pathSupplier) {
-        return GWT.getModuleBaseURL() + "defaulteditor/download?path=" + pathSupplier.get().toURI();
+
+        return GWT.getModuleBaseURL() + "defaulteditor/download?path=" + URIUtil.encodeQueryString(URIUtil.decode(pathSupplier.get().toURI()));
```

Comment:
"defaulteditor/download?path=" can we make this literal  as a constant

Reply:
Reasonable suggestion and done.